Merge branch 'main' into prototype/signal-forms

leonsenft opened the pull request.

AndrewKushnir committed:
refactor(core): include DI path into cyclic dependency error message (#50902)

This commit updates the logic to better handle a situation when there is a cyclic DI dependency detected. Previously, the error message used to contain the name of the token that triggered the problem. With this change, the DI resolution path would also be included, so that it's easier to find and resolve the cycle.

PR Close #50902

ivanwonder committed:
feat(language-service): support to report the deprecated API in the template (#62054)

In the Typescript Language Service, these diagnostics are reported as suggestion diagnostics.
This will report the deprecated `Component`, `Directive`, etc.

Fixes https://github.com/angular/angular/issues/59343

PR Close #62054

RafaelJCamara committed:
feat(compiler-cli): add diagnostic for uninvoked functions in text interpolation (#59191)

This commit adds a new extended diagnostic to detect uninvoked functions in text interpolation.

PR Close #59191

JeanMeche committed:
fix(devtools): do not emit a route object with functions (#62254)

In the case where a Route has title defined as a function, it eventually throw when it reaches `window.postMessage`

fixes #60595

PR Close #62254

milomg committed:
refactor(devtools): add cookies signal demo (#62749)

add a new component that includes a small signal graph and explicitly calls .set on a signal.

PR Close #62749

milomg committed:
refactor(core): use quad linked lists for signals (#62284)

inspired by the design of Preact's signals, use linked lists instead of arrays for faster signal creation

PR Close #62284

milomg committed:
fix(core): update symbols for new signals api (#62284)

approve the new symbols that appear from bundling the signals changes

PR Close #62284

JeanMeche committed:
docs(docs-infra): generate llms-full.txt at build time (#61864)

PR Close #61864

AleksanderBodurri committed:
test(devtools): revive cypress tests (#61972)

Previously these tests would run automatically when Angular DevTools lived in another repo. These files have continued to live here but have not been running automatically on each PR.

Now, these test files have been revived to run properly with our changes since the repo merge. This is a first step to reviving our e2e testing.

Next steps include writing cypress tests for new features like Injector Graph, Router tree, signals visualizations, etc.

PR Close #61972

JeanMeche committed:
docs: improve support for decorator APIs (#60411)

The commit introduce the distinction between

- class like decorators (like Component, Interface, NgModule. They are formatted like classes/interfaces, with each attribute being documents.
- function like decorators (Attribute, Host, Optional...)

PR Close #60411

dgp1130 committed:
refactor(devtools): remove `version_name` from `manifest.json` (#62351)

This field defaults to `version`, so there's no need to have it when it's exactly the same. This is one less number to bump during release PRs.

PR Close #62351

MeAkib committed:
docs(docs-infra): replace deprecated RouterTestingModule with RouterModule.forRoot([]) (#62364)

PR Close #62364

thePunderWoman committed:
refactor(core): add compiler support for animation instructions (#62528)

this adds the compiler code to support the animate instructions.

PR Close #62528

josephperrott committed:
build: update to latest @angular/build-tooling (#62380)

Update to the latest @angular/build-tooling package, which drop dependency on @angular/benchpress

PR Close #62380

milomg committed:
fix(devtools): fix some build errors from g3 typescript (#62412)

this avoids some issues with casting types

PR Close #62412

hawkgs committed:
refactor(devtools): remove leftover code from router-tree template (#62396)

Remove some SVG visualization leftover code from the template.

PR Close #62396

hawkgs committed:
refactor(devtools): make tree visualizations accessible (#62396)

Improve the accessibility of the SVG tree visualizations by employing `<title>` and `aria-labelledby`.

PR Close #62396

Commit:
docs: fix dynamic form reactivity (#62400)

PR Close #62400

Commit:
docs: fix formatting (#62400)

PR Close #62400

milomg committed:
refactor(devtools): update signal body on change (#62410)

when the epoch of a signal changes, update the preview text

PR Close #62410

josephperrott committed:
build: migrate adev to use packaged version of @angular/* packages (#62413)

Use the packaged versions of the packages instead of the local ts_project dependencies to prevent multiple versions of the deps to enter test bundles

PR Close #62413

josephperrott committed:
build: migrate devtools to use packaged version of @angular/* packages (#62413)

Use the packaged versions of the packages instead of the local ts_project dependencies to prevent multiple versions of the deps to enter test bundles

PR Close #62413

SkyZeroZx committed:
refactor(platform-browser): use RuntimeError for hydration feature conflicts (#62414)

Replace generic Error with RuntimeError

PR Close #62414

SkyZeroZx committed:
docs: Update documentation httpResource with credentials option (#62415)

PR Close #62415

SkyZeroZx committed:
feat(http): Add support for fetch referrer & integrity options in HttpClient (#62417)

Enhances Angular HttpClient requests with Fetch API referrer and integrity options to optimize security

PR Close #62417

JeanMeche committed:
docs(docs-infra): Load of the typing files from Angular. (#62424)

Following some infra changes, DTS files are now splitted over several files, which aren't referenced in the `package.json` but are directly imported in the root dts.

We chose to not use ATA here because it only pulls the latest version from a CDN whereas we prefer to use the local definitions we have in the `node_modules` of the tutorial.

fixes #62374

PR Close #62424

JeanMeche committed:
docs: display deprecation message for top most symbol (#62427)

For classes, types, functions, enums etc.

fixes #62365

PR Close #62427

crisbeto committed:
perf(core): move property remapping for dom properties to compiler (#62421)

Since we know that DOM properties won't go to an inputs, we can move the remapping logic to the compiler, saving us some processing on the client.

PR Close #62421

angular-robot committed:
build: update all non-major dependencies (#62419)

See associated pull request for more information.

PR Close #62419

devversion committed:
refactor(compiler-cli): avoid eager `typescript` top-level usages (#62431)

For some internal refactorings, we want to avoid any direct top-level
`typescript` runtime usages. Usages that interact with `ts` after
synchronous module loading are fine.

More context in:
https://docs.google.com/document/d/1_h8AQRbM3LJrwIlUfGzGzxuwqFu0QI2-5eUTFI6gWws/edit?tab=t.0#heading=h.nunsebe2s3h0.

PR Close #62431

devversion committed:
refactor(compiler-cli): do not rely on `url` when loading file system logic (#62431)

When executing the compiler-cli in the browser, or some parts of it—
modules like `url` are stubbed out with `{}`. This commit accounts for
this.

PR Close #62431

angular-robot committed:
build: update cross-repo angular dependencies to v20.1.0-rc.0 (#62433)

See associated pull request for more information.

PR Close #62433

josephperrott committed:
build: migrate adev to use rules_angular based cli rules (#62435)

Use the @angular/cli rules from rules_angular instead of the npm package directly

PR Close #62435

MeAkib committed:
refactor(docs-infra): cleanup lines in scss (#62438)

Cleans up some of the scss code.

PR Close #62438

MeAkib committed:
refactor(docs-infra): removed unused imports (#62441)

Cleans up instances of unused imports in a few shared docs components.

PR Close #62441

josephperrott committed:
build: migrate remaing esbuild and nodejs_binary usages to new toolchain (#62435)

Migrate the remaing usages of esbuild and nodejs_binary in adev to the new toolchain versions.

PR Close #62435

josephperrott committed:
build: only require workspace build packages for web tests when provided (#62443)

Only use the builds from the workspace as for the bootstrap scripts for web testing

PR Close #62443

josephperrott committed:
build: migrate all dependencies for adev into adev/package.json (#62435)

Migrate the dependencies into the adev/package.json instead of importing from the the root package.json

PR Close #62435

devversion committed:
refactor(migrations): ensure tsurge can properly emit references in g3 (#62447)

Currently when Tsurge runs in g3, it creates a bare bones Angular
compiler plugin. The tsconfigs from compilation units may set options
like "useHostForImportGeneration", but the Ngtsc logic doesn't enable
because the `fileNameToModuleName` method is not defined on the host.

This can break reference emission for Tsurge analyzers/programs and
result in subtle differences to real `ng_module` compilations. This
commit fixes this by making the method available in 1P Tsurge.

Notably, reference emission can occur during analysis— so even if
migrations aren't "emitting TS -> JS" output.

PR Close #62447

thePunderWoman committed:
ci: update pullapprove reviewers (#62448)

This sets some people to be reviewers, but not actively requested to account for availability.

PR Close #62448

thePunderWoman committed:
test: rename legacy animations integration tests (#62527)

This just prefixes the integration tests associated with the animation packages.

PR Close #62527

JeanMeche committed:
refactor(devtools): remove the standalone demo (#62455)

The default demo is already standalone

PR Close #62455

sEver committed:
fix(docs-infra): fix tutorial formatting issues (for v20) (#62457)

Fixes including, but not limited to:
- `visibleLines` values
- all codeblocks normalized to 2 spaces indentation
- lesson 14-http sEverely de-indented
- descriptions and explanations were rewritten to reflect the changes from @Input to input and from ngFor to @for
- missing step added in lesson 11-details-page
- rewritten confusing instructions for adding code that was already added in previous steps

PR Close #62457

SkyZeroZx committed:
feat(http): add referrer & integrity support for fetch requests in httpResource (#62461)

This commit adds support for the Fetch API's referrer & integrity options when using httpResource with the withFetch provider

PR Close #62461

SkyZeroZx committed:
docs: Update documentation http with integrity & referrer (#62462)

PR Close #62462

angular-robot committed:
build: update dependency mermaid to v11.8.0 (#62463)

See associated pull request for more information.

PR Close #62463

mderazon committed:
docs: fix typo on guidelines file (#62468)

Correct a minor spelling mistake in the documentation to improve clarity.

PR Close #62468

hawkgs committed:
refactor(devtools): unify table styles (#62466)

Make tabular data styles consistent across the app, whether that's a native or Material table.

PR Close #62466

angular-robot committed:
build: update dependency aspect_rules_ts to v3.6.2 (#62475)

See associated pull request for more information.

PR Close #62475

crisbeto committed:
fix(compiler-cli): use proper name for diagnostic type (#62479)

Currently when a diagnostic happens in a template, we always say "Error occurs in ...", however we have other types of diagnostics as well.

PR Close #62479

dzhavat committed:
docs: update comment (#62482)
PR Close #62482

SkyZeroZx committed:
refactor(http): use RuntimeError for invalid timeout error (#62490)

Replace generic Error with RuntimeError

PR Close #62490

devversion committed:
refactor: fix closure compatibility errors in compiler (#62493)

The static fields may be part of advanced compilations where the
receiver would change as part of the static field collapsing
optimization.

The optimization requires us to use the explicit class name, over
`this` that would change to e.g. `globalThis`.

PR Close #62493

devversion committed:
refactor(compiler-cli): ensure mock file system can be loaded in the browser (#62493)

The mock file system currently relies on some NodeJS logic. This blocks
the convenient use of the mock file system in the browser. The file
system is useful for creating tests to verify e.g. that our transforms
can work in the browser.

PR Close #62493

arturovt committed:
refactor(service-worker): align `getRegistration()` types (#62497)

In this commit, we align the service worker types with those provided by TypeScript. The `ServiceWorkerContainer.getRegistration()` method returns a `Promise<ServiceWorkerRegistration | undefined>`, but we were casting it to `Observable<ServiceWorkerRegistration>`, ignoring the possibility of `undefined`.

This is unsafe, as we should account for the case where the registration is `undefined` and handle it appropriately in the push manager.

PR Close #62497

josephperrott committed:
build: migrate angular-optimization from devinfra to devtools (#62503)

Migrate angular-optimization to devtools instead of loading via npm package

PR Close #62503

devversion committed:
refactor(compiler): ensure compiler passes closure conformance checks (#62500)

This commit adjusts some code of the compiler that currently results in
conformance checks failures with JSCompiler/Closure.

PR Close #62500

dimakuba committed:
fix(router): ensure preloaded components are properly activated (#62502)

Preloaded components were not being activated in certain scenarios when preloading was enabled. This change ensures that components are correctly activated after being preloaded.

PR Close #62502

angular-robot committed:
build: update all non-major dependencies (#62476)

See associated pull request for more information.

PR Close #62476

josephperrott committed:
build: disable firefox tests in devtools (#62504)

Disabling firefox tests in devtools until we can get them working on OS X as expected

PR Close #62504

josephperrott committed:
build: migrate modules to use dependencies from //modules:package.json (#62506)

Migrate to use local package.json deps

PR Close #62506

bencodezen committed:
docs: add dedicated routing lifecycle and events guide (#62225)

doc: apply suggestions from code review

Co-authored-by: Enea Jahollari <jahollarienea14@gmail.com>

Co-authored-by: Matthieu Riegler <kyro38@gmail.com>

PR Close #62225

angular-robot committed:
build: lock file maintenance (#62494)

See associated pull request for more information.

PR Close #62494

angular-robot committed:
build: update cross-repo angular dependencies (#62507)

See associated pull request for more information.

PR Close #62507

angular-robot committed:
build: update dependency aspect_rules_ts to v3.6.3 (#62512)

See associated pull request for more information.

PR Close #62512

devversion committed:
refactor(compiler): fix import manager to be closure property renaming-compatible (#62517)

The `generateUniqueIdentifier` helper relies on the internal `identifiers` property of a `ts.SourceFile`.
As this is not a public-facing API, it is not included in the externs provided to Closure Compiler.
Consequently, it is susceptible to being renamed during advanced optimizations, which would lead to runtime failures.

To prevent this, the property is now accessed via a string literal (`sf['identifiers']`). This change ensures that
the property name is preserved through the compilation process. An explanatory comment has been added to clarify the
necessity of this approach for future reference.

PR Close #62517

JeanMeche committed:
docs(docs-infra): fix banner position on mobile & tablet (#62521)

#62509

PR Close #62521

josephperrott committed:
build: migrate usages of protractor_web_test_suite to use new rules_browser based rule (#62525)

Use the protractor_web_test_suite from rules_browser instead of @bazel/protractor

PR Close #62525

JeanMeche committed:
docs(docs-infra): update update guide when changing level (#62526)

This commit also fixes the colors of the material button toggle

fixes #62432

PR Close #62526

josephperrott committed:
build: migrate esbuild usages to aspect_rules_esbuild esbuild rule (#62529)

Migrate more usages of esbuild to new toolchain

PR Close #62529

JeanMeche committed:
refactor(devtools): remove unused imports (#62530)

Import had no purpose and was also invalid.

PR Close #62530

JeanMeche committed:
docs(docs-infra): run analytics only on the browser (#62531)

We don't care about server, adev only uses SSG.
Also `setCookieConsent` is throwing on node.

PR Close #62531

JeanMeche committed:
docs(docs-infra): do not cache the versions.json file (#62534)

Without the flag, the file gets inline in the transfer cache making the resource useless.

PR Close #62534

arturovt committed:
fix(router): handle errors from view transition readiness (#62535)

This commit adds a `.catch()` handler to `transition.ready` from `document.startViewTransition` to prevent `AbortError`s in Safari when `startViewTransition` is called synchronously multiple times.

PR Close #62535

leonsenft committed:
feat(core): render ARIA property bindings as attributes (#62630)

Allow binding to ARIA attributes using property binding syntax _without_
the `attr.` prefix. For example, `[aria-label]="expr"` is now valid, and
equivalent to `[ariaLabel]="expr"`. Both examples bind to either a
matching input or the `aria-label` HTML attribute, rather than the
`ariaLabel` DOM property.

Binding ARIA properties as attributes will ensure they are rendered
correctly on the server, where the emulated DOM may not correctly
reflect ARIA properties as attributes.

Reuse the DOM schema registry from the compiler to map property names in
type check blocks.

PR Close #62630

alan-agius4 committed:
fix(http): do not display warnings `Angular detected that a `HttpClient` request with the `keepalive` option was sent using XHR` when option is not true (#62536)

Currently, this warning is always displayed as the option defaults to false. We update the logic to only display the warning when it's true.

PR Close #62536

crisbeto committed:
feat(core): support TypeScript 5.9 (#62541)

Updates the repo to support TypeScript 5.9 and expands the allowed version range.

PR Close #62541

hawkgs committed:
refactor(devtools): show profiler progress bar during recording (#62547)

Show the profiler progress bar during the full recording instead of only when there are no frames.

PR Close #62547

JeanMeche committed:
refactor(devtools): use signal inputs for split component (#62550)

This also introduced a local direction signal so it can be written from the outside

PR Close #62550

dgp1130 committed:
release: bump DevTools version to 1.0.38 (#62552)

PR Close #62552

ice-blaze committed:
docs(forms): remove unecessary `FormGroup` casting (#62555)

Fixes #62554

PR Close #62555

josephperrott committed:
build: use rollup directory instead of rollup through @bazel/rollup (#62556)

Use rollup directly instead of using the previous toolchain implementation

PR Close #62556

josephperrott committed:
build: use tsc from @npm2// instead of @npm// (#62558)

Use tsc from the new node_modules layout

PR Close #62558

mmalerba committed:
ci: add gemini code assist config (#62560)

Configures gemini to not interact with new pull requests, only review on
demand

PR Close #62560

AndrewKushnir committed:
docs: release notes for the v20.1.0 release

AndrewKushnir committed:
release: cut the v20.2.0-next.0 release

angular-robot committed:
docs: update Angular CLI help (#62520)

Updated Angular CLI help contents.

PR Close #62520

josephperrott committed:
build: use esbuild from aspect rules (#62568)

Use the esbuild ruleset from aspect rules instead of via @bazel/esbuild

PR Close #62568

milomg committed:
refactor(devtools): simplify vendored deps to make importing into google3 easier (#62567)

prefer using node_modules/webtreemap, and remove memo-decorator

PR Close #62567

thePunderWoman committed:
feat(core): add enter and leave animation instructions (#62682)

This adds the instructions to support enter and leave animations on nodes.

PR Close #62682

alan-agius4 committed:
ci: update renovate label for non main branch (#62570)

The non default base branch is no longer in rc phase

PR Close #62570

alan-agius4 committed:
docs: update browsers support versions (#62572)

The current URL is not correct as it incorrectly includes browsers. We now replace the URL with the comparer provided by baseline which will be available soonish via `web-platform-dx.github.io`

PR Close #62572

hawkgs committed:
refactor(devtools): improve breadcrumbs component (#62575)

Drop the redundant and lefover code. Slightly update the look.

PR Close #62575

hawkgs committed:
refactor(devtools): cleanup Material leftovers (#62575)

Cleanup some Material leftovers and drop some redundant markup.

PR Close #62575

JeanMeche committed:
build: set paths for devtools & adev (#62574)

This allows the IDE to find the right imports

PR Close #62574

hawkgs committed:
fix(devtools): make font sizes consistent (#62575)

Fix the inconsistent font sizes that are result of Material changes.

PR Close #62575

hawktuahyuri committed:
docs: fix comment breaking Highlighter (#62577)

This fixes a comment that was causing issues with docs highlighting

PR Close #62577

angular-robot committed:
build: update all non-major dependencies (#62513)

See associated pull request for more information.

PR Close #62513

thePunderWoman committed:
ci: adjust renovate teams (#62578)

This switches framework team to approve but not requested, and adds the currently active caretaker group to the teams for renovate.

PR Close #62578

wartab committed:
fix(http): Add missing timeout and transferCache options to `HttpClient` (#62586)

The `timeout` option is implemented in the different HTTP backends, but the option is not actually passed to the backends. `transferCache` is also missing in one signature. This commit adds both options to `HttpClient`.

Fixes #62581

PR Close #62586

JeanMeche committed:
refactor(compiler): Error on comment only interpolations (#62590)

This commit introduces a ParserError to prevent an error later in the pipepine

fixes #34084

PR Close #62590

thePunderWoman committed:
ci: add to public-api approvers list (#62591)

This appends reviewers to the approvers list.

PR Close #62591

lizclipse committed:
fix(core): fix change tracking for Resource#hasValue (#62595)

When using `hasValue()` I would expect it to behave like any other
reactive value such that changes to the internal `value()` that do not
cause `hasValue()` to return anything different do not trigger change
detection, but this was not the case. This change wraps the value
checking in a `computed` such that it behaves as expected again while
still preserving the type narrowing.

PR Close #62595

angular-robot committed:
build: update all non-major dependencies (#62582)

See associated pull request for more information.

PR Close #62582

JeanMeche committed:
docs: split devtools doc to prepare for upcomming updates (#62597)

PR Close #62597

angular-robot committed:
docs: update Angular CDK apis (#62600)

Updated Angular CDK api files.

PR Close #62600

JeanMeche committed:
test(docs-infra): Use the zoneless testing rule (#62596)

This will spare use the warning logs when the tests run.

```
      NG0914: The application is using zoneless change detection, but is still loading Zone.js. Consider removing Zone.js to get the full benefits of zoneless. In applications using the Angular CLI, Zone.js is typically included in the "polyfills" section of the angular.json file.
```

PR Close #62596

fosfor15 committed:
docs(docs-infra): fix nested routes example (#62608)
PR Close #62608

JeanMeche committed:
docs: fix CLI reference layout (#62609)

fixes #62602

PR Close #62609

JeanMeche committed:
refactor(devtools): Use OnPush on all components (#62610)

The end goal is to move the devtools app to zoneless.

PR Close #62610

JeanMeche committed:
docs: add note about difference between class bindings and  `ngClass` (#62611)

fixes #40623

PR Close #62611

forivall committed:
docs: update forgotten change to input signals (#62612)
PR Close #62612

JeanMeche committed:
docs(docs-infra): Add status to some guide entries. (#62614)

Guide entries can now have "new" or "updated" tags.

PR Close #62614

aparzi committed:
fix(core): InputBinding marks component a dirty. (#62613)

This fix ensures that CD runs on OnPush component when an input binding is set.

PR Close #62613

joaodasilva committed:
docs: fix lazy load example in define-routes.md (#62620)

Fixes #62619

PR Close #62620

angular-robot committed:
build: update all non-major dependencies (#62603)

See associated pull request for more information.

PR Close #62603

josephperrott committed:
build: move all rule/macro loading into devtools/defaults.bzl (#62627)

Move all of our loading into a local defaults.bzl file instead of mixed loading via tools/defaults and devtolls/defaults

PR Close #62627

josephperrott committed:
build: move all rule/macro loading into adev/defaults.bzl and fix intellisense loading (#62627)

Move all of our loading into a local defaults.bzl file instead of mixed loading via tools/defaults and adev/defaults

PR Close #62627

muhammadali1658 committed:
docs: fix missing 'of' in Style Guide sentence (#62628)

PR Close #62628

atscott committed:
fix(core): Ensure application remains unstable during bootstrap (#62631)

This commit ensures the application remains unstable during the entire
bootstrap process. This ensures all bootstrap listeners and app
initializers observe the application as being unstable until each one
has gotten a chance to execute the synchronous block (potentially adding
more pending tasks).

Prior to this commit, application initializers or bootstrap
listeners may observe the application as being stable, even though other
initializers/listeners had not yet executed. This created an ordering
issue whereby the hydration bootstrap listener would observe the
application as stable prior to the router performing its initial
navigation.

fixes #62592

PR Close #62631

josephperrott committed:
docs: correctly look for examples in the bin_dir instead of execPath (#62632)

Properly look for the example files for live previews in the bin_dir

PR Close #62632

JeanMeche committed:
refactor(devtools): replace/remove animations from the animation package (#62633)

Animations can be handled by the styling directly

PR Close #62633

bencodezen committed:
docs: add dedicated data resolvers router guide (#62250)

PR Close #62250

JeanMeche committed:
refactor(devtools): Migrate the devtools app to zoneless. (#62454)

This commit also removed the useless ngZone invocations

PR Close #62454

josephperrott committed:
build: migrate protractor test to new toolchain (#62649)

Migrate to new toolchain for protractor test in packages/examples/upgrade

PR Close #62649

josephperrott committed:
build: remove extra toolchain registration (#62649)

Remove extra toolchain registration that is redundant as we already add the toolchains elsewhere

PR Close #62649

josephperrott committed:
build: remove rules_nodejs based protractor tooling (#62649)

Remove the protractor setup based on rules_nodejs

PR Close #62649

SkyZeroZx committed:
docs: Update documentation httpResource with referrer & integrity options (#62637)

PR Close #62637

crisbeto committed:
docs: update best practices prompt (#62640)

Updates the `best-practices.md` with results from recent testing.

PR Close #62640

crisbeto committed:
fix(compiler): incorrect spans for left side of binary operation (#62641)

Fixes that the span for the `left` side of a `Binary` AST included the range up to and including the operator.

Fixes #62617.

PR Close #62641

crisbeto committed:
test(compiler): remove duplicated lexer tests (#62644)

There were 26 duplicated block tests in `lexer_spec.ts`, likely due to merge conflicts. These changes remove the duplicates while keeping the 6 tests that were different.

PR Close #62644

crisbeto committed:
fix(compiler): more permissive parsing of @ characters (#62644)

When we introduced blocks, we made a deliberate decision to treat the `@` character as a reserved character in case we need to use it for other syntax in the future. This meant that some common cases, like writing out an email address in the template, can be broken.

After some recent discussions we decided to relax the requirement and only treat `@` as a reserve character if it's followed by a character sequence that matches a known block.

PR Close #62644

crisbeto committed:
refactor(compiler): indicate whether element is void at AST level (#62648)

Updates the HTML AST to indicate whether a specific element is a void element.

PR Close #62648

angular-robot committed:
build: update all non-major dependencies (#62642)

See associated pull request for more information.

PR Close #62642

crisbeto committed:
refactor(core): expose function to assert type of variables (#62648)

Adds the `ɵassertType` function that will be used during type checking to assert the type of a value.

PR Close #62648

josephperrott committed:
ci: use new size tracking in integration tests (#62650)

Use the new size tracking within integration tests

PR Close #62650

crisbeto committed:
fix(compiler-cli): infer type of event target for void elements (#62648)

Currently we infer the target of DOM events to be `EventTarget | null` which is consistent with the built-in types for `addEventListener`. This is due to the fact that users can dispatch custom events, or the event might've bubbled. However, this typing is also inconvenient for some other common use cases like `<input (input)="query($event.target.value)">`, because we don't have the ability to type cast in a template.

These changes aim to make some of the cases simpler by inferring the type of `$event.target` if the event is bound on a void element which guarantees that it couldn't have bubbled.

PR Close #62648

hawkgs committed:
refactor(devtools): restyle the signal details panel (#62681)

Restyle the details panel to match and be inline with the existing UI/style.

PR Close #62681

JeanMeche committed:
refactor(devtools): simular transfer cache data in the devtools demo app. (#62665)

With the commit we mimic server-side behavior and serialize a bunch of transfer cache data.

fixes #62664

PR Close #62665

JrPribs committed:
docs: fixes binding box typo in httpResource.value() template example (#62661)

PR Close #62661

tonypconway committed:
docs: Update versions.md to use canonical Baseline site (#62680)

replace `tonypconway` fork URL with canonical `web-platform-dx` URL

PR Close #62680

JeanMeche committed:
fix(compiler): fix detection of directive deps in JIT (#62666)

This fix also matches the implementation to the jsdoc for `hasDirectiveDependencies` "Whether any of the component's dependencies are directives"

fixes #62573

PR Close #62666

AndrewKushnir committed:
docs: release notes for the v20.1.1 release

AndrewKushnir committed:
release: cut the v20.2.0-next.1 release

hybrist committed:
release: bump Angular DevTools version to 1.0.39 (#62670)

PR Close #62670

angular-robot committed:
build: update all non-major dependencies (#62662)

See associated pull request for more information.

PR Close #62662

josephperrott committed:
build: migrate zone.js to use ts_project instead of ts_library (#62673)

Use ts_project instead of ts_library for building zone.js

PR Close #62673

josephperrott committed:
build: remove ts_library definition (#62673)

Remove the ts_library macro from our defaults.bzl file

PR Close #62673

atscott committed:
refactor(core): Use promise.finally for task removal (#62674)

Updates the approach from https://github.com/angular/angular/pull/62631 to
use Promise.finally for task removal. This is a bit cleaner and also handles
rejections from the donePromise

PR Close #62674

josephperrott committed:
build: migrate ts_library usages to ts_project (#62673)

Migrate new ts_library usages to ts_project

PR Close #62673

SkyZeroZx committed:
feat(http): Add redirected property to HttpResponse and HttpErrorResponse (#62675)

Add support for the Fetch API's redirected property in HttpResponse and HttpErrorResponse when using HttpClient with the withFetch provider.

The redirected property indicates whether the response was the result of an HTTP redirect, providing valuable information for security, debugging, and conditional logic.

PR Close #62675

angular-robot committed:
build: update dependency node to v22.17.1 (#62676)

See associated pull request for more information.

PR Close #62676

milomg committed:
refactor(devtools): disable view source button if source function is not available (#62678)

if there is no debuggableFn, we shouldn't allow users to click the view source button

PR Close #62678

hawkgs committed:
refactor(devtools): rearrange main toolbar menu buttons (#62684)

Place the version in the info menu and move both the info and gear buttons to the right-hand side of the toolbar.

PR Close #62684

hawkgs committed:
refactor(devtools): make the component metadata menu more prominent (#62685)

Add an arrow icon to indicate that this is an expansion panel. Reduce the active area (which toggles the menu) to the text only.

PR Close #62685

josephperrott committed:
build: migrate to new toolchain usage for api goldens (#62688)

Migrate api golden usage to be based on rules_js toolchain implementation

PR Close #62688

josephperrott committed:
build: remove unused rules/macros from defaults.bzl (#62709)

Remove rules/macros that are not longer used in the repo

PR Close #62709

angular-robot committed:
build: update cross-repo angular dependencies (#62557)

See associated pull request for more information.

PR Close #62557

NothingEverHappens committed:
docs: release notes for the v20.1.2 release

hawkgs committed:
fix(devtools): make some containers scrollable (#62703)

Add scrolling to some containers in the app.

PR Close #62703

hawkgs committed:
refactor(devtools): introduce docs-ref-button (#62685)

Used for external Angular docs links.

PR Close #62685

geromegrignon committed:
docs(router): rephrase the package installation (#62704)
PR Close #62704

bampakoa committed:
docs: fix minor typos in styleguide (#62706)
PR Close #62706

Avcharov committed:
feat(devtools): add transfer state tab (#62465)

Add transfer state tab, which is taking transfer state script by using APP_ID. Created internal api ɵgetTransferState to retrieve transfer state value from app into devtools app.

PR Close #62465

JeanMeche committed:
docs: guard a resource error with `hasValue()` (#62708)

fixes #62687

PR Close #62708

josephperrott committed:
build: move nodejs_test to js_test (#62709)

Move nodejs_test to js_test and remove unnecessary files after move

PR Close #62709

josephperrott committed:
build: migrate tslint rules from dev-infra to local (#62709)

Migrate the tslint rules used in dev-infra to locally defined rules as they are unused in other repos

PR Close #62709

josephperrott committed:
build: remove unused @bazel/* npm packages (#62709)

Remove the unused bazel npm packages

PR Close #62709

JeanMeche committed:
fix(devtools): add event tagging to prevent DDOS (#62645)

One common problem encountered by the devtools content script is that it accepted almost any message send over the message bus. Some websites like `auth.openai.com` were spamming the bus and DDOS the devtools app.

By introducing event tagging and skipping non-devtools events we prevent DDOS of the Angular devtools content script by on forward tagged events.

PR Close #62645

wbreakell committed:
docs: update link to git fixup documentation (#62717)

PR Close #62717

angular-robot committed:
build: update all non-major dependencies (#62701)

See associated pull request for more information.

PR Close #62701

SkyZeroZx committed:
feat(service-worker): notify clients about version failures (#62718)

Add client notification when an app version fails, improving error
visibility and debugging capabilities. When a version encounters a
critical error, all clients using that version are now notified
with details about the failure.

- Add notifyClientsAboutVersionFailure method call in versionFailed
- Ensure clients receive VERSION_FAILED events with error details
- Improve service worker error transparency for better debugging

PR Close #62718

jhuang007007 committed:
docs: fix code snippet (#62720)

changes code snippet on first-app tutorials (https://angular.dev/tutorials/first-app/09-services) for the step "Add a new file level import for the HousingService:" from 

imports: [HousingLocation], 

to

import {HousingService} from '../housing.service';
PR Close #62720

SkyZeroZx committed:
feat(service-worker): Adds support for updateViaCache in provideServiceWorker (#62721)

Enables configuration of the updateViaCache setting for service worker registration, allowing greater control over caching behavior during updates. Extends tests to cover new option and ensures correct propagation in registration logic.

PR Close #62721

JeanMeche committed:
refactor(core): add a `getTransferState` for the devtools (#62722)

`getTransferState` will expose public data from the transfer state. It will for example remove internal hydration data.

PR Close #62722

AleksanderBodurri committed:
test(devtools): bump cypress timeout to 10 seconds (#62726)

By default this is 4 seconds. In CI we have unreliable runtime performance so we this change gives us a bit of wiggle room to validate application behaviour.

PR Close #62726

AleksanderBodurri committed:
fix(devtools): stop reseting currentlyMatchedIndex when a node is selected in the component explorer (#62727)

This was causing some unintended behaviour when paired with our new zoneless configuration. I'm not sure why currentlyMatchedIndex was getting set to -1 here. With this removed it seems like things are working as expected. Previously a select call would reset this index *after* a search filter was applied, which would cause weird behaviour with the "next" and "prev" buttons.

PR Close #62727

angular-robot committed:
build: update cross-repo angular dependencies (#62728)

See associated pull request for more information.

PR Close #62728

angular-robot committed:
docs: update Angular CLI help (#62729)

Updated Angular CLI help contents.

PR Close #62729

thePunderWoman committed:
refactor(compiler): Tree shake Element Registry (#62682)

This creates a feature to detect usages of animate.leave and only enables the element removal registry when necessary

PR Close #62682

josephperrott committed:
build: remove all usages of the interop_deps attr for ts_project and ng_project (#62732)

Remove all of the usages of interop_deps as attributes in the repo

PR Close #62732

josephperrott committed:
build: move generating doc files for @angular/ssr to using an intermediate typescript file (#62734)

Use an intermediate typescript file to rexport all of all of @angular/ssr rather than reaching into the npm package files

PR Close #62734

SkyZeroZx committed:
feat(service-worker): Improves storage full detection in data caching (#62737)

Adds proactive storage capacity checks using the Storage API
to better handle near-full conditions during data caching.
Replaces previous TODOs with actual usage monitoring and
logging to help prevent cache failures due to quota exhaustion.

PR Close #62737

angular-robot committed:
build: update github/codeql-action action to v3.29.3 (#62739)

See associated pull request for more information.

PR Close #62739

crisbeto committed:
fix(compiler): allow more characters in square-bracketed attribute names (#62742)

Currently the HTML parser will stop parsing as soon as it hits an end character in the name of an attribute (e.g. `/` or `>`). This ends up being problematic with some third-party packages like Tailwind which uses a wider range of characters for its class names. While the characters are fine when inside the `class` attribute, our current parser behavior prevents users from setting those classes conditionally through `[class.]` bindings.

These changes adjust the parser to handle such cases.

Fixes #61671.

PR Close #62742

angular-robot committed:
build: update @angular/ng-dev digest to 75a2aee (#62744)

See associated pull request for more information.

PR Close #62744

angular-robot committed:
build: update dependency angular-split to v20 (#62740)

See associated pull request for more information.

PR Close #62740

clydin committed:
docs: create MCP server setup guide (#62746)

Introduces a new guide for setting up the Angular CLI's Model Context Protocol (MCP) server.

The guide includes:
- Instructions on how to use the command.
- IDE-specific JSON configurations for VS Code, JetBrains, Firebase Studio, and Cursor.
- Clarification on the different JSON structures required by each IDE.

PR Close #62746

JeanMeche committed:
docs: add mentions of dynamic title resolver / strategy (#62751)

PR Close #62751

crisbeto committed:
fix(platform-browser): resolve component resources before bootstrapping in JIT mode (#62758)

Currently if the app has some external resources and it gets bootstrapped in JIT mode, the user will get an error about calling `resolveComponentResources`. These changes add a call to `resolveComponentResources` automatically inside `bootstrapApplication`.

Note that we don't need similar logic for `bootstrapModule`, because it calls into `compileNgModuleFactory` internally which in turn calls `resolveComponentResources`.

PR Close #62758

crisbeto committed:
refactor(core): expose function for inferring tag name (#62283)

Exposes the code that we use to infer the tag name of a component as a private export so that we can use it in `TestBed`.

PR Close #62283

crisbeto committed:
feat(core): add option to infer the tag names of components in tests (#62283)

Currently when testing a component using `TestBed.createComponent`, we always create the component as a `div` which isn't aligned with the runtime. The runtime tries to parse out the tag name from the first selector in `@Component` and only falls back to `div` if there isn't one. This behavior difference can cause components to not behave like they would in production which reduces the usefulness of the tests.

These changes add the `inferTagName` option to `TestBed.createComponent` and `TestBed.configureTestingModule` that allows apps to opt into inferring the tag name from the selector in the same way as the runtime. Currently the new option is set to `false`, but we intend to change it to `true` in a future version.

PR Close #62283

thePunderWoman committed:
ci: update renovate reviewers (#62761)

This restores the framework-team availability, since the setting was not working as intended.

PR Close #62761

thePunderWoman committed:
refactor(core): add configurable behavior for animations to testbed (#62764)

This adds a test module configuration to define whether animations should be enabled or disabled in test. By default, they are disabled.

PR Close #62764

sbarfurth committed:
fix(http): propagate plain errors when parsing fails (#62765)

The fetch backend now propagates the plain body when parsing the body fails.
This replicates the behavior of the XHR backend introduced in #19773.

The current state completely obfuscates errors of the "wrong" response type.
However, it's not uncommon for successful requests to return one type and
errors to return another type. Propagating the plain error allows downstream
error consumers to reason about the error body and decide how to parse it
depending on application needs.

PR Close #62765

hybrist committed:
release: bump Angular DevTools version to 1.0.40 (#62767)

PR Close #62767

thePunderWoman committed:
fix(core): checks if body exists before continuing (#62768)

This fixes a case in which the document exists, but the body does not.

PR Close #62768

hybrist committed:
docs: call out cleanup steps in devtools release process (#62769)

PR Close #62769

atscott committed:
build: update bazel rules_browsers dependency (#62772)

Updates only rules_browsers since
https://github.com/angular/angular/pull/62542 is having difficulty with
other deps updated at the same time

PR Close #62772

thePunderWoman committed:
fix(core): switch check to documentElement with chaining (#62773)

This fixes a situation where DOM checks may report null.

PR Close #62773

josephperrott committed:
build: move from using rollup to esbuild for bundling zone.js (#62775)

Use esbuild to bundle zone.js

PR Close #62775

SkyZeroZx committed:
refactor(forms): Removes redundant destroy ref usage in value accessor in select control (#62738)

Simplifies destruction logic by relying directly on the injector's destroyed state.
Eliminates unnecessary retrieval of a separate destroy reference

PR Close #62738

WhoisBsa committed:
docs(docs-infra): fix indentation in documentation (#62774)

fix minor indentation issue in a code block to improve readability
PR Close #62774

atscott committed:
test(router): Use autoTick feature in router tests (#62776)

This updates async router tests to use the jasmine autoTick feature. Observed test timings
for the chromium tests went down from ~7.5s to ~3.5-4s. For node, these decreased from ~5.5s to ~3s.

In addition to the speed improvement, this feature:

* Removes the need to be careful about timeout ordering when there are several timeouts in tests.
* Removes the need to ensure test timeouts are kept sufficiently small
* Reduces overall flakiness

PR Close #62776

josephperrott committed:
build: remove local zone.js build from usage in integration tests (#62775)

Remove the local zone.js build from dependencies in the integration tests

PR Close #62775

clydin committed:
docs: add initial documentation for MCP server tools (#62777)

Adds documentation for the tools available in the Angular CLI MCP server. This includes:
- get_best_practices
- search_documentation
- list_projects

Also includes a section for providing feedback and ideas for new capabilities.

PR Close #62777

NothingEverHappens committed:
docs: release notes for the v20.1.3 release

NothingEverHappens committed:
release: cut the v20.2.0-next.2 release

angular-robot committed:
build: update cross-repo angular dependencies (#62770)

See associated pull request for more information.

PR Close #62770

milomg committed:
refactor(devtools): cleanup iframe-message-bus (#62785)

remove unused function and cleanup private readonly properties

PR Close #62785

milomg committed:
refactor(devtools): cleanup a few devtools bazel files (#62786)

remove unused styles, update to es2022

PR Close #62786

angular-robot committed:
build: update all non-major dependencies (#62754)

See associated pull request for more information.

PR Close #62754

angular-robot committed:
build: update github/codeql-action action to v3.29.4 (#62789)

See associated pull request for more information.

PR Close #62789

angular-robot committed:
build: lock file maintenance (#62793)

See associated pull request for more information.

Closes #62621 as a pr takeover

PR Close #62793

thePunderWoman committed:
refactor(animations): deprecate the animations package (#62795)

This deprecates the animations package in favor of using `animate.enter` and `animate.leave` with intent to remove the full package in v22.2.

DEPRECATED: @angular/animations

PR Close #62795

GarikHk committed:
docs(docs-infra): correct typos in adev and wording in resource api (#62801)
PR Close #62801

niklas-wortmann committed:
docs: fix issue in jetbrains MCP configuration (#62802)

PR Close #62802

josephperrott committed:
build: migrate zone.js tests to use web test runner instead of karma web tests (#62826)

Migrate zone.js bazel tests to use web test runner instead of karma

PR Close #62826

josephperrott committed:
build: remove all usages of @bazel/runfiles (#62804)

Remove usages of @bazel/runfiles throughout the repository

PR Close #62804

josephperrott committed:
build: use ts_config rule from rules_ts only and always load from common location (#62808)

Only use the ts_config rule from rules_ts, and always load the rule from tools/defaults2.bzl

PR Close #62808

atscott committed:
build: Export language service bundle (#62811)

fixes Plugin activation failed: Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './bundles/language-service.js' is not defined by 'exports'

PR Close #62811

SkyZeroZx committed:
docs: Update VersionFailedEvent in Service Worker docs (#62813)

PR Close #62813

angular-robot committed:
build: update cypress-io/github-action action to v6.10.2 (#62814)

See associated pull request for more information.

PR Close #62814

crisbeto committed:
build: update to the TypeScript 5.9 RC (#62815)

Updates the repo the release candidate of TypeScript 5.9.

PR Close #62815

angular-robot committed:
build: update cross-repo angular dependencies (#62806)

See associated pull request for more information.

PR Close #62806

cexbrayat committed:
docs: fix link for NG8117 uninvokedFunctionInTextInterpolation (#62820)

PR Close #62820

josephperrott committed:
build: use rollup to locally bundle domino (#62823)

Use rollup instead of esbuild for bundling domino locally for usage within the repository

PR Close #62823

josephperrott committed:
build: move tsec to new toolchain (#62825)

Migrate tsec tooling/macro to use tsec directly as a js_binary and rely on ts_project dependencies.

PR Close #62825

JeanMeche committed:
docs: restore mention of structural directives. (#62836)

The link was removed in #61708, but we should actually kept it but point to the dedicated page

fixes #62835

PR Close #62836

JeanMeche committed:
refactor(common): replace aio links to adev (#62839)

The is a doc change + a minor test change (to remove test logs that mention aio)

PR Close #62839

cexbrayat committed:
fix(compiler-cli): display proper function in NG8117 message (#62842)

The diagnostic was displaying `'Function in text interpolation should be invoked: ect Object]()` instead of `'Function in text interpolation should be invoked: firstName()'.

PR Close #62842

AleksanderBodurri committed:
test(devtools): write some e2e tests for the transfer state tab (#62843)

Tests validate the current behaviour of the transfer state feature:

- Tab can be made visible by enabling in the settings menu
- Transfer state component displays keys total and size
- Validates the existence of the mock content present in https://github.com/angular/angular/tree/main/devtools/src/app/transfer-state.ts

PR Close #62843

Avcharov committed:
refactor(docs): rename idx to firebase studio (#62848)

Rename IDX, as since now IDX is a part of Firebase Studio

PR Close #62848

angular-robot committed:
build: lock file maintenance (#62849)

See associated pull request for more information.

PR Close #62849

hawkgs committed:
fix(devtools): transfer state settings menu item text size (#62851)

Fix the text size of the transfer state settings menu item. Also, rename "Show" to "Enable" to match the rest of the non-default tabs text.

PR Close #62851

JeanMeche committed:
fix(compiler): exclude more safe reads expression from 2way-binding (#62852)

Priori to this fix the parser would allow safereads in accestor receievers (but the direct one).

fixes #62837

PR Close #62852

JeanMeche committed:
docs: fix ActivateRoute example (#62855)

PR Close #62855

angular-robot committed:
build: update cross-repo angular dependencies (#62856)

See associated pull request for more information.

PR Close #62856

josephperrott committed:
build: standardize usage of js_binary (#62860)

Remove usages of nodejs_binary and use a commonly defined js_binary macro

PR Close #62860

thePunderWoman committed:
Revert "test(devtools): write some e2e tests for the transfer state tab (#62843)" (#62867)

This reverts commit 3a1a6cc20deb10fe790a6d8137c5b7b95c3ddf17.

PR Close #62867

angular-robot committed:
build: update all non-major dependencies (#62862)

See associated pull request for more information.

PR Close #62862

thePunderWoman commented:
@leonsenft You should be just rebasing that branch rather than creating a new commit for it.

thePunderWoman reviewed (changes requested):
I'm happy to assist with a full rebase of the signal-forms branch.

hawkgs committed:
fix(devtools): truncate text in the property tab header (#62868)

Handle very long component names when the "Show Signal Graph" button is visible.

PR Close #62868

kyjus25 committed:
docs: fix const for pi calculation (#62872)

PR Close #62872

josephperrott committed:
build: use ts_project based extract_types (#62875)

Use the extract_types that pulls from ts_project deps

PR Close #62875

angular-robot committed:
build: update cross-repo angular dependencies (#62876)

See associated pull request for more information.

PR Close #62876

josephperrott committed:
build: remove final usages of nodejs_binary macros (#62878)

Remove the remaining usages of our nodejs_binary macros

PR Close #62878

josephperrott committed:
build: remove unused macro (#62878)

Remove unused npm_package_bin macro

PR Close #62878

josephperrott committed:
build: move http_server and generate_api_docs into defaults2.bzl (#62878)

Move http_server and generate_api_docs into defaults2.bzl as they are rules_js compliant

PR Close #62878

josephperrott committed:
build: remove the esm interop patches and tooling (#62879)

Remove esm interop patches and tooling from previous toolchain

PR Close #62879

leonsenft committed:
Merge remote-tracking branch 'upstream/main' into prototype/signal-forms
